### DISCUSSION TITLE
React - Delay group context performances

### DISCUSSION THREAD

**Discussion by gsouf** [floating-ui/floating-ui#2669] [General]
On the website is documented how to use a delay group https://floating-ui.com/docs/tooltip#delay-groups

This example uses the `FloatingDelayGroupContext` provided by floating ui to offer shared delay accross various tooltips.

I want to provide a single shared context for my whole app by wrapping it all under a single `FloatingDelayGroup`.  However as it's using React's context (via `useDelayGroupContext`), then I would expect it to rerender every tooltips whenever I activate one of the items.

After testing, it does not seem to be the case. It doesn't seem to rerender everything.

 Is there anything special in the implementation that I missed and that prevents all tooltips using the same context to rerender everytime?

Thanks!

**Comment by atomiks** [comment]
The state hooks changing inside `FloatingDelayGroup` don't affect the whole children tree due to this principle: https://twitter.com/sebmarkbage/status/1096115287781400576

Only the consumers (so `Tooltip` component) end up re-rendering.

**Reply by gsouf** [reply]
That's good to know thanks!
